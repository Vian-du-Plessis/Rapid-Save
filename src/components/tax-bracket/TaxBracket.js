import React, { useEffect, useState } from 'react';
import styles from './TaxBracket.module.scss';

const TaxBracket = (props) => {

    const [value, setValue] = useState([{name: 'Please add income'}]);

    useEffect(() => {
        setValue(props.value);
    }, [props.value])

    return (
        <div className={ styles.container }>
            <h3>Tax Brackets</h3>
            <hr />
            {
                value.map((item, index) => 
                    <div key={index}>
                        <h4><strong>{item.name}</strong></h4>
                        <h5>{item.message}</h5>
                    </div>
                )
            }
        </div>
    );
};

export default TaxBracket;