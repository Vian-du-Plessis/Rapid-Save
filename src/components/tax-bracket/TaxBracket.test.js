import TaxBracket from './TaxBracket';
import { render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

const taxObject = [{
    name: 'Vian',
    message: 'Vian does not pay any tax'
}]

const TaxBracketProps = {
    title: 'Household income summary',
    value: taxObject
}

describe("Test Income Summary Component... ", () => {

    test("Show correct title...", () => {
        render(
            <TaxBracket
                {...TaxBracketProps}
            />
        );
        const titleElement = screen.getByText("Tax Brackets");

        expect(titleElement).toBeInTheDocument();
        expect(titleElement.textContent).toBe("Tax Brackets");
    });

    test("Show correct sub Heading One...", () => {
        render(
            <TaxBracket
                {...TaxBracketProps}
            />
        );
        const titleElement = screen.getByText("Vian does not pay any tax");

        expect(titleElement).toBeInTheDocument();
        expect(titleElement.textContent).toBe("Vian does not pay any tax");
    });

    // SNAPSHOT TEST
    test("Snapshot of component", () => {
        const component = renderer.create(
            <TaxBracket/>
        );

        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        tree = component.toJSON();
        expect(tree).toMatchSnapshot()
    })

})