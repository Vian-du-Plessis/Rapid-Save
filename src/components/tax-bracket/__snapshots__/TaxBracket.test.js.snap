// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Income Summary Component... Snapshot of component 1`] = `
<div
  className="container"
>
  <h3>
    Tax Brackets
  </h3>
  <hr />
  <div>
    <h4>
      <strong>
        Please add income
      </strong>
    </h4>
    <h5 />
  </div>
</div>
`;

exports[`Test Income Summary Component... Snapshot of component 2`] = `
<div
  className="container"
>
  <h3>
    Tax Brackets
  </h3>
  <hr />
  <div>
    <h4>
      <strong>
        Please add income
      </strong>
    </h4>
    <h5 />
  </div>
</div>
`;
